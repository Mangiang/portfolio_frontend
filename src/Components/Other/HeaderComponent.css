:local(.text){
    font-family: "Open Sans", sans-serif;
    color: black;
}

:local(.centerName) {
    composes: text;
    text-align: left;
    font-size: 40px;
}

:local(.title){
    margin: 0;
}

:local(.subTitle) {
    composes: text;
    font-size: 25px;
    padding: 15px 0 0 0;
}

:local(.centerDescription) {
    composes: text;
    text-align: left;
    font-size: 20px;
}

:local(.divLine) {
    margin: 20px 0 0 0;
    width: 60%;
    background-color: black;
    height: 3px;
}