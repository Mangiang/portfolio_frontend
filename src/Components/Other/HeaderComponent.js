import React, {Component} from 'react';
import {hot} from 'react-hot-loader';
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import style from './HeaderComponent.css';

type Props = {
    requestNavigation: (currentPage: string) => any,
    history: Object
}

export class HeaderComponent extends Component<Props> {
    constructor(props: Props) {
        super(props);
    }

    render() {
        return (
            <div id="title" className={style['centerName']}>
                <h2 className={style['title']}>Arthur Joly</h2>
                <div id="subTitle" className={style['subTitle']}>Computer science engineer student</div>
                <div className={style['divLine']}/>
            </div>
        )
    }
}

export const HeaderComponentConnected = connect()(HeaderComponent);
export default hot(module)(withRouter(HeaderComponentConnected));
