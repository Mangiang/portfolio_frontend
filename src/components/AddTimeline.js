import React from 'react';
import {hot} from 'react-hot-loader';

import {connect} from 'react-redux';
import {addTimeline} from '../actions/actions';


class AddTimeline extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputTitle: "",
            inputDescription: "",
            beginDate: new Date(),
            endDate: new Date()
        };

        this.onClickAdd = this.onClickAdd.bind(this);
        this.updateInputDescription = this.updateInputDescription.bind(this);
        this.updateInputTitle = this.updateInputTitle.bind(this);
        this.updateBeginDate = this.updateBeginDate.bind(this);
        this.updateEndDate = this.updateEndDate.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClickAdd() {
        this.props.dispatch(addTimeline(
            this.state.inputTitle,
            this.state.inputDescription,
            this.state.beginDate,
            this.state.endDate,
            this.state.token
        ));
        this.setState({
            inputDescription: "",
            inputTitle: "",
            beginDate: new Date(),
            endDate: new Date()
        })      
    }

    onClickHide() {
        
    }

    updateInputTitle(e) {
        this.setState({
            inputTitle: e.target.value
        })
    }

    updateInputDescription(e) {
        this.setState({
            inputDescription: e.target.value
        })
    }
    
    updateBeginDate(date) {
        this.setState({
            beginDate:date 
        })
    }
    
    updateEndDate(date) {
        this.setState({
            endDate:date 
        })
    }

    render() {
        if (this.state.token != "none")
            return (
                <div id="projectForm">
                    <form>
                        <h3>New timeline</h3>
                        <div className="form-group">
                            <label htmlFor="titlet">Title</label>
                            <input className="form-control" type="text" name="titlet" id="titlet" onChange={this.updateInputTitle} value={this.state.inputTitle}/>
                        </div>
                        <div className="form-group"> 
                            <label htmlFor="description">Description</label>
                            <input className="form-control" type="description" name="description" id="description" onChange={this.updateInputDescription} value={this.state.inputDescription}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="beginDate">Begining date</label>
                        </div>
                        <div className="form-group">
                            <label htmlFor="endDate">Ending date</label>
                        </div>
                        <button type="button" className="btn btn-default" onClick={() => this.onClickAdd()}>Add</button>
                    </form>                
                </div>
            );
        else
            return <div></div>
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;

    return {
        token
     };
}

export default hot(module)(connect(mapStateToProps)(AddTimeline));