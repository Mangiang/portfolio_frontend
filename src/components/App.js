import React from 'react'
import {hot} from 'react-hot-loader'

import {connect} from 'react-redux';

import {getProjects} from '../actions/actions';
import ProjectsListScreen from '../screens/ProjectsListScreen';

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            projects: []
        }
    }

    componentWillReceiveProps(nextProps) {
        this.state.projects = nextProps.projects;
    }

    componentWillMount() {
        this.props.dispatch(getProjects());
        this.state.projects = this.props.projects;
    }

    render() {
        return(
            <ProjectsListScreen/>
        )
    }
}

function mapStateToProps(state) {
    const projects = state.projects;
    console.log("asdasd");
    console.log(state.projects);

    return {
        projects
     };
}

export default hot(module)(connect(mapStateToProps)(App));