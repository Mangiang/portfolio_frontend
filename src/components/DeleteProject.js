import React from 'react';
import { hot } from 'react-hot-loader';
import DatePicker from 'react-datepicker';
import moment from 'moment';

import { connect } from 'react-redux';
import { deleteProject } from '../actions/actions';

import 'react-datepicker/dist/react-datepicker.css';

class DeleteProject extends React.Component {
    constructor(props) {
        super(props)
        this.state = {

        }

        this.onClick = this.onClick.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClick() {
        this.props.dispatch(deleteProject(this.props.id, this.props.token));
    }

    render() {
        if (this.state.token != "none")
            return (
                <div className="deleteForm">
                    <form>
                        <button type="button" className="btn btn-default" onClick={() => this.onClick()}>X</button>
                    </form>                
                </div>
            );
        else
            return <div></div>
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;

    return {
        token
     };
}

export default hot(module)(connect(mapStateToProps)(DeleteProject));