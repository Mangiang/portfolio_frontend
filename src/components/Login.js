import React from 'react';
import { hot } from 'react-hot-loader';

import { connect } from 'react-redux';
import { login, logout } from '../actions/actions';



class Login extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            inputLogin: "",
            inputPassword: "",
            token: "none"
        }

        this.onClickLogin = this.onClickLogin.bind(this);
        this.updateInputLogin = this.updateInputLogin.bind(this);
        this.updateInputPassword = this.updateInputPassword.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.state.token = nextProps.token;
        if (nextProps.loginFailed)
            this.blinkForm("#CC0000");
    }

    componentWillMount() {
        this.state.token = this.props.token;
    }

    onClickLogin() {
        this.props.dispatch(login(this.state.inputLogin, this.state.inputPassword));      
    }

    onClickLogout() {
        this.props.dispatch(logout());
    }

    blinkForm(color) {
        let elt = document.getElementById("loginForm").childNodes[0];
        elt.style.backgroundColor = color;
    }

    updateInputLogin(e) {
        this.setState({
            inputLogin: e.target.value
        })
    }

    updateInputPassword(e) {
        this.setState({
            inputPassword: e.target.value
        })
    }

    render() {
        if (this.state.token == "none")
            return (
                <form className="form-inline" id="loginForm">
                    <div className="form-group">
                        <label htmlFor="login">Login</label>
                        <input className="form-control" type="text" name="login" id="login" onChange={this.updateInputLogin} />
                    </div>
                    <div className="form-group"> 
                        <label htmlFor="password">Password</label>
                        <input className="form-control" type="password" name="password" id="password" onChange={this.updateInputPassword} />
                    </div>
                    <button type="button" className="btn btn-default" onClick={() => this.onClickLogin()}>Log in</button>
                </form>             
            );
        else
            return (
                <div id="logoutButton">
                    <button type="button" className="btn btn-default" onClick={() => this.onClickLogout()}>Log out</button>
                </div>
            );
    }
}

function mapStateToProps(state) {
    const token = state.loginInfos.token;
    const loginFailed = state.loginInfos.loginFailed;
    console.log(token + "  miaou")

    return {
        token,
        loginFailed
     };
}

export default hot(module)(connect(mapStateToProps)(Login));