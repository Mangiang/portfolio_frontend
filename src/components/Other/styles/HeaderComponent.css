:local(.centerName) {
    color: white;
}

:local(.centerDescription) {
    margin-top: 50px;
    color: white;
}
