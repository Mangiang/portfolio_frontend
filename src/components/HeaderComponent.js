// @flow

import React, {Component} from 'react';
import {hot} from 'react-hot-loader';
import {connect} from 'react-redux';
import {withRouter} from 'react-router';
import type {Dispatch} from "redux"

import {styles} from './styles/HeaderComponentStyles'


import Parallax from 'react-parallax';

import {requestNavigation} from '../actions/actions';

type Props = {
    currentPage: string,
    requestNavigation: (currentPage: string) => any,
    history: Object
}

class HeaderComponent extends Component<Props> {
    constructor(props) {
        super(props);
    }

    render() {
        let description = "";
        if (this.props.currentPage === 'timeline')
            description = "Here are both my professional and school my experience";
        else if (this.props.currentPage === 'projectsList')
            description = "Here is a list of my projects";

        return (
            <div>
                <Parallax
                    blur={3}
                    bgImage={require('../images/backgroundImage.jpg')}
                    bgImageAlt="the background image"
                    strength={500}>
                    <div style={styles.backgroundOverlay}>
                        <h1 style={styles.centerName}>Arthur Joly's portfolio</h1>
                        <h4 style={styles.centerDescription}>
                            {description}
                            <br/>
                        </h4>
                        <button onClick={this.handleTimelineRedirect.bind(this)}>
                            {(this.props.currentPage !== 'timeline' ? 'Timeline' : 'Projects list')}
                        </button>
                        <div style={{height: 200}}/>
                    </div>
                </Parallax>
            </div>
        )
    }

    handleTimelineRedirect() {
        if (this.props.currentPage === "timeline")
            this.props.requestNavigation('projectsList');
        else {
            this.props.requestNavigation('timeline');
        }
        this.props.history.push('/' + this.props.currentPage);
    }
}

function mapStateToProps(state) {
    const destinationPage = state.navHeader;
    if (!destinationPage || Object.keys(destinationPage).length === 0) {
        return {
            currentPage: 'projectsList'
        };
    }
    return {
        currentPage: destinationPage
    };
}

function mapDispatchToProps(dispatch: Dispatch<*>) {
    return {
        requestNavigation: (newPage) => {
            dispatch(requestNavigation(newPage));
        }
    }
}

export default hot(module)(withRouter(connect(mapStateToProps, mapDispatchToProps)(HeaderComponent)));