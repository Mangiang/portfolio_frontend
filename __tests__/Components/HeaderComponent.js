import React from 'react'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'

import {HeaderComponent} from '../../src/components/HeaderComponent'

// Snapshot for MyHeaderComponent React Component
describe('>>> HEADER COMPONENT', () => {
    let wrapper;
    const currentPage = "The current page";
    const history = {};
    const requestNavigation = () => "requestNavigation";

    beforeEach(() => {
        wrapper = shallow(<HeaderComponent currentPage={""} history={{}} requestNavigation={() => "toto"}/>)
    });

    describe('--- Snapshot', () => {
        it('+++capturing Snapshot of MyHeaderComponent', () => {
            const renderedValue = renderer.create(wrapper).toJSON();
            expect(renderedValue).toMatchSnapshot();
        });
    });
});
