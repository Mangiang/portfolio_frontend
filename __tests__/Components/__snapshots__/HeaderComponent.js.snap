// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> HEADER COMPONENT --- Snapshot +++capturing Snapshot of MyHeaderComponent 1`] = `
<div>
  <div
    className="sc-bwzfXH jcxrUP"
  >
    <div
      className="sc-htpNat VxUFJ sc-bdVaJa bkBxAm"
      height={null}
      src="backgroundImage.jpg"
      style={
        Object {
          "backgroundPositionY": "center",
        }
      }
      width={null}
    />
    <div
      className="sc-bxivhb fEUpac sc-bdVaJa bkBxAm"
      color="#4F4943"
      opacity={0.5}
    >
      <div>
        <h1
          style={
            Object {
              "color": "white",
              "paddingTop": 50,
            }
          }
        >
          Arthur Joly's portfolio
        </h1>
        <h4
          style={
            Object {
              "color": "white",
              "marginTop": 50,
            }
          }
        >
          
          <br />
        </h4>
        <button
          onClick={[Function]}
        >
          
        </button>
      </div>
    </div>
  </div>
</div>
`;
